platform/linux/src/headless_display_egl.cpp

```diff
@@ -0,0 +1,64 @@
+#include <mbgl/platform/default/headless_display.hpp>
+#include <mbgl/util/string.hpp>
+
+#include <EGL/egl.h>
+
+namespace mbgl {
+
+class HeadlessDisplay::Impl {
+public:
+    Impl();
+    ~Impl();
+
+    EGLDisplay display = 0;
+    EGLConfig config = 0;
+};
+
+HeadlessDisplay::Impl::Impl() {
+    display = eglGetDisplay(EGL_DEFAULT_DISPLAY);
+    if (display == EGL_NO_DISPLAY) {
+        throw std::runtime_error("Failed to obtain a valid EGL display.\n");
+    }
+
+    EGLint major, minor, numConfigs;
+    if (!eglInitialize(display, &major, &minor)) {
+        throw std::runtime_error("eglInitialize() failed.\n");
+    }
+
+    const EGLint attribs[] = {
+        EGL_SURFACE_TYPE, EGL_DONT_CARE,
+        EGL_RED_SIZE, 1,
+        EGL_GREEN_SIZE, 1,
+        EGL_BLUE_SIZE, 1,
+        EGL_ALPHA_SIZE, 0,
+        EGL_DEPTH_SIZE, 1,
+        EGL_RENDERABLE_TYPE, EGL_OPENGL_BIT,
+        EGL_NONE
```

Comment:
Maybe add a note that the configuration here doesn't matter since we're rendering to a framebuffer anyway.